ssl/ssl_privkey.cc

```diff
@@ -386,7 +386,16 @@ static bool ssl_public_key_rsa_pss_check(EVP_PKEY *pubkey, uint16_t sigalg) {
 bool ssl_public_key_supports_signature_algorithm(SSL_HANDSHAKE *hs,
                                                  uint16_t sigalg) {
   SSL *const ssl = hs->ssl;
-  if (!pkey_supports_algorithm(ssl, hs->local_pubkey.get(), sigalg)) {
+  // We may have a public key that supports the signature algorithm, but the
+  // negotiated cipher needs to also allow it. This behavior is only done prior
+  // to TLS 1.2 servers in OpenSSL since TLS 1.3 does not have cipher-based
+  // authentication configuration.
```

Comment:
nit: Might be worth making a note that the TLS 1.3 variant is checked via the `SSL_aGENERIC` bit flag being set. Since all those ciphers specify this value. Assuming my understanding of this code flow is correct.

Reply:
Yes that was the intention behind using `SSL_aGENERIC`. No problem, definately worth making a note of.